src/sid/contacts.py

```diff
@@ -329,39 +344,43 @@ def _calculate_infections_by_recurrent_contacts(
                 # extract infection probability into a variable for faster access
                 base_probability = infection_probs[cm]
                 for j in others:
-                    # the case i == j is skipped by the next if condition because it
-                    # never happens that i is infectious but not immune
-                    if not immune[j] and recurrent_contacts[j, cm] > 0:
+                    j_is_susceptible = not infectious[j] and newly_infected[j] == -1
+                    if j_is_susceptible and recurrent_contacts[j, cm] > 0:
                         # j is infected depending on its own susceptibility.
                         virus_strain_i = virus_strain[i]
                         contagiousness_factor_i = contagiousness_factor[virus_strain_i]
+                        immunity_resistance_factor_i = immunity_resistance_factor[
+                            virus_strain_i
+                        ]
+
                         individual_infection_risk = (
                             base_probability
                             * susceptibility_factor[j]
                             * contagiousness_factor_i
+                            * (1 - immunity_resistance_factor_i * immunity[j])
```

Comment:
The larger the immunity resistance factor, the lower the chance of infection? :raised_eyebrow:

Reply:
Changed to (1 - (1 - ``immunity_resistance_factor_i``) * ``immunity[j]``).